src/plugins/intel_gpu/include/intel_gpu/runtime/memory_pool.hpp

```diff
@@ -162,6 +162,8 @@ class memory_pool {
     void dump(uint32_t id, uint32_t iter, std::string dump_dir_path = "");
     size_t get_total_mem_pool_size(allocation_type type);
 
+    static size_t count_users_in_padded_pool(memory_pool& pool, cldnn::memory_ptr& mem_ptr);
```

Comment:
From what I understood, reset is also required for a single-layer usage where shape changes. For example, if shape is reduced from 224x224 with pad=2, to 223x223 with pad=2. Am I missing something here?

Reply:
@jade-cho  It is not the matter of getting memory from memory pool. 
1. iter N ;  224x224 + pad2 => allocate mem 
2. iter N+1 : 200x200 + pad2 => not allocate mem (i.e., not getting a new memory from pool. It is using the assigned memory as-is.) Here the padding area is dirty due to the previous run.

Reply:
So I don't think this is an essential fix for the possible cases.

Reply:
@yeonbok In this case, `can_reuse_buffer` becomes true, so it is not related to the issue I am trying to solve.